### PR TITLE
Routes

### PR DESCRIPTION
**Pull request by jaylanjones421**

### CODE REVIEW

File: client/src/Components/PrivateRoute/PrivateRoute.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, { Component } from "react";
+import { withRouter, Route, Redirect } from "react-router-dom";
+import { connect } from "react-redux";
+
+import HomeView from "../HomeView/HomeView";
+
+class PrivateRoute extends Component {
+  render() {
+    let protectedComponent = props => {
```
Comment (donovanhiland) [inline client/src/Components/PrivateRoute/PrivateRoute.js]: I don't think this needs to be a function. The render method already is a function and you have access to props

File: client/src/Components/PrivateRoute/PrivateRoute.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      let { userRole, userID, component, tickets, path } = props;
```
Comment (donovanhiland) [inline client/src/Components/PrivateRoute/PrivateRoute.js]: you can just change this to `this.props` and you'll be just fine without the wrapping function

File: client/src/Components/PrivateRoute/PrivateRoute.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      if (userRole && userRole === "Owner") {
+        return <Route path={path} component={component} />;
+      } else if (userRole && userRole === "Worker") {
+        if (tickets.find(x => x.worker_id === userID)) {
+          return <Route path={path} component={HomeView} />;
```
Comment (donovanhiland) [inline client/src/Components/PrivateRoute/PrivateRoute.js]: Is the home view the same place you're redirecting to in the else at the end?

Comment (jaylanjones421) [reply client/src/Components/PrivateRoute/PrivateRoute.js]: homeview is the list of tickets, the redirect at the end is to the login view

**Comment by donovanhiland** [comment]
das nice das real nice

**Pull request merged**